internal/juju/machines.go

```diff
@@ -252,21 +297,49 @@ func (c machinesClient) ReadMachine(input ReadMachineInput) (ReadMachineResponse
 	if !exists {
 		return response, fmt.Errorf("no status returned for machine: %s", input.ID)
 	}
+	c.Tracef("ReadMachine:Machine status result", map[string]interface{}{"machineStatus": machineStatus})
 	response.ID = machineStatus.Id
-	channel, err := series.ParseChannel(machineStatus.Base.Channel)
+	response.Base, response.Series, err = baseAndSeriesFromParams(&machineStatus.Base)
 	if err != nil {
 		return response, err
 	}
-	// This might cause problems later, but today, no one except for juju internals
-	// uses the channel risk. Using the risk makes the base appear to have changed
-	// with terraform.
-	response.Base = fmt.Sprintf("%s@%s", machineStatus.Base.Name, channel.Track)
-	response.Series = machineStatus.Series
 	response.Constraints = machineStatus.Constraints
-
 	return response, nil
 }
 
+// readMachineWithRetryOnNotFound calls ReadMachine until
+// successful, or the count is exceeded when the error is of type
+// not found. Delay indicates how long to wait between attempts.
+func (c machinesClient) readMachineWithRetryOnNotFound(ctx context.Context, input ReadMachineInput) (ReadMachineResponse, error) {
+	var output ReadMachineResponse
+	err := retry.Call(retry.CallArgs{
+		Func: func() error {
+			var err error
+			output, err = c.ReadMachine(input)
+			typedErr := typedError(err)
+			if errors.Is(typedErr, errors.NotFound) {
+				return nil
+			}
+			return err
+		},
+		NotifyFunc: func(err error, attempt int) {
+			if attempt%4 == 0 {
+				message := fmt.Sprintf("waiting for machine %q", input.ID)
+				if attempt != 4 {
+					message = "still " + message
+				}
+				c.Debugf(message)
+			}
+		},
+		BackoffFunc: retry.DoubleDelay,
+		Attempts:    30,
```

Comment:
is 30 attempts a little too many?

Reply:
30 is arbitrary for sure. In practicality, it's a max time of 61 sec due to the backoff func of DoubleDelay and wait time of 1 sec. 

We also have the ctx.Done() that's passed through from the terraform client as a backstop.